app/frontend/components/pages/dashboard/dashboardPage.tsx

```diff
@@ -22,201 +22,210 @@ import AccountsDashboard from '../accounts/accountsDashboard'
-  accountsTabs = ['Accounts']
-  stakingTabs = ['Delegate ADA', 'Current Delegation', 'Staking history']
-  sendingTabs = ['Send ADA', 'Transactions', 'Receive ADA']
-  advancedTabs = ['Keys', 'Certificate']
-  render({selectedMainTab}, {selectedSubTab}) {
-    const selectedDefultSubTabs = {
-      Accounts: 'Accounts',
-      Sending: 'Transactions',
-      Staking: 'Delegate ADA',
-      Advanced: 'Keys',
-    }
-    const tabs = {
-      Accounts: this.accountsTabs,
-      Sending: this.sendingTabs,
-      Staking: this.stakingTabs,
-      Advanced: this.advancedTabs,
-    }
-    if (!tabs[selectedMainTab].includes(selectedSubTab)) {
-      this.selectSubTab(selectedDefultSubTabs[selectedMainTab])
-    }
-    const Page = this.pages[selectedSubTab]
-    return (
-      <div className="dashboard-content">
-        <ul className="dashboard-tabs">
-          {tabs[selectedMainTab].map((name, i) => (
-            <SubTab
+const DashboardPage = ({
+  selectedMainTab,
+  displayInfoModal,
+  isShelleyCompatible,
+  shouldShowNonShelleyCompatibleDialog,
+  shouldShowPremiumBanner,
+  shouldShowSaturatedBanner,
+  activeAccountIndex,
+  shouldNumberAccountsFromOne,
+  selectMainTab,
+}: Props) => {
+  return (
+    <div className="page-wrapper">
+      <ErrorModals />
+      {isShelleyCompatible && displayInfoModal && <InfoModal />}
+      {shouldShowNonShelleyCompatibleDialog && <NotShelleyCompatibleDialog />}
+      {!isShelleyCompatible && <DashboardErrorBanner />}
+      {shouldShowPremiumBanner && <PremiumBanner />}
+      {shouldShowSaturatedBanner && <SaturationErrorBanner />}
+      {ADALITE_CONFIG.ADALITE_CARDANO_VERSION === 'shelley' && (
+        <ul className="tabinator">
+          {Object.values(MainTabs).map((name, i) => (
+            <MainTab
               key={i}
               name={name}
-              selectedTab={selectedSubTab}
-              selectTab={this.selectSubTab}
+              isSelected={name === selectedMainTab}
+              selectTab={() => selectMainTab(name)}
+              displayName={
+                name === MainTabs.ACCOUNT &&
+                (shouldNumberAccountsFromOne
+                  ? `Account #${activeAccountIndex + 1}`
+                  : `Account ${activeAccountIndex}`)
+              }
             />
           ))}
         </ul>
-        <Page />
-      </div>
-    )
-  }
+      )}
+      {MainPages[selectedMainTab]}
+    </div>
+  )
+}
+
+type DashboardMobileProps = {
+  subTabs: SubTabs[]
+  defaultSubTab: SubTabs
+  mainSubTab?: SubTabs
+}
+
+const DashboardMobileContent = ({subTabs, defaultSubTab, mainSubTab}: DashboardMobileProps) => {
+  const [selectedSubTab, selectSubTab] = useState(defaultSubTab)
+  return (
+    <div className="dashboard-content">
+      {mainSubTab && SubPages[mainSubTab]}
+      <ul className="dashboard-tabs">
+        {subTabs.map((name, i) => (
+          <SubTab
+            key={i}
+            name={name}
+            isSelected={name === selectedSubTab}
```

Comment:
Also I'd rename selectedSubTab and selectedMainTab everywhere to active[Main/Sub]Tab, it's super distracting right now

Reply:
I ll change it to `active` and `setActive`, good point :+1: